modules/plottinggl/plotters/persistencediagramplotgl.h

```diff
@@ -0,0 +1,161 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#ifndef IVW_PERSISTENCEDIAGRAMPLOTGL_H
+#define IVW_PERSISTENCEDIAGRAMPLOTGL_H
+
+#include <modules/plottinggl/plottingglmoduledefine.h>
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/datastructures/transferfunction.h>
+#include <inviwo/core/properties/compositeproperty.h>
+#include <inviwo/core/properties/ordinalproperty.h>
+#include <inviwo/core/properties/boolproperty.h>
+#include <inviwo/core/properties/transferfunctionproperty.h>
+#include <inviwo/core/interaction/pickingmapper.h>
+
+#include <modules/opengl/texture/textureutils.h>
+#include <modules/opengl/shader/shader.h>
+#include <modules/base/algorithm/dataminmax.h>
+
+#include <modules/plotting/datastructures/dataframe.h>
+#include <modules/plotting/properties/marginproperty.h>
+#include <modules/plotting/properties/axisproperty.h>
+#include <modules/plottinggl/utils/axisrenderer.h>
+
+namespace inviwo {
+
+class Processor;
+using IndexBuffer = Buffer<std::uint32_t, BufferTarget::Index>;
+class PickingEvent;
+
+namespace plot {
+
+/**
+ * \class PersistenceDiagramPlotGL
+ * \brief base class for plotting a persistence diagram using OpenGL
+ */
+class IVW_MODULE_PLOTTINGGL_API PersistenceDiagramPlotGL {
+public:
+    class Properties : public CompositeProperty {
+    public:
+        InviwoPropertyInfo();
+
+        Properties(std::string identifier, std::string displayName,
+                   InvalidationLevel invalidationLevel = InvalidationLevel::InvalidResources,
+                   PropertySemantics semantics = PropertySemantics::Default);
+
+        Properties(const Properties &rhs);
+        Properties &operator=(const Properties &that);
+        virtual Properties *clone() const override;
+        virtual ~Properties() = default;
+
+        BoolProperty showPoints_;
+        FloatProperty radius_;
+        FloatProperty lineWidth_;
+        FloatProperty lineWidthDiagonal_;
+        FloatVec4Property pointColor_;
+        FloatVec4Property lineColor_;
+        FloatVec4Property hoverColor_;
+        TransferFunctionProperty tf_;
+        MarginProperty margins_;
+        FloatProperty axisMargin_;
+
+        FloatProperty borderWidth_;
+        FloatVec4Property borderColor_;
+
+        BoolProperty hovering_;
+
+        AxisProperty xAxis_;
+        AxisProperty yAxis_;
+    };
+
+    PersistenceDiagramPlotGL(Processor *processor);
```

Comment:
Could use a callback to remove dependency on processor?

Reply:
wish that I could. The PickingMapper relies on the processor for handling picking events.